apps/protocol-frontend/src/components/DaoDashboardShell.tsx

```diff
@@ -1,24 +1,110 @@
+import { useState } from 'react';
 import { Box, Flex, Heading } from '@chakra-ui/react';
+import { ControlledSelect, ControlledDatePicker } from '@govrn/protocol-ui';
 import PageHeading from '../components/PageHeading';
 import ContributionTypesPieShell from './ContributionTypesPieShell';
 import RecentContributionsTableShell from './RecentContributionsTableShell';
+
+import { subWeeks } from 'date-fns';
 interface DaoDashboardShellProps {
   daoName: string;
   daoId: number;
 }
-
-const startDate = new Date('7/1/2022'); // temporary hardcoded value while working on the datepicker component
-const endDate = new Date('8/1/2022'); // temporary hardcoded value while working on the datepicker component
+const TODAY_DATE = new Date();
 
 const DaoDashboardShell = ({ daoName, daoId }: DaoDashboardShellProps) => {
+  const [showCustomDatePicker, setShowCustomDatePicker] = useState(false);
+
+  const [startDate, setStartDate] = useState<Date>(new Date(TODAY_DATE));
+  const [endDate, setEndDate] = useState<Date>(
+    new Date(subWeeks(TODAY_DATE, 4)),
+  );
+  const [dateRange, setDateRange] = useState<{
+    label: string;
+    value: number;
+  } | null>(null);
+
+  const dateRangeOptions = [
+    {
+      value: 'Custom',
+      label: 'Custom',
+    },
+    { value: 1, label: 'Last Week' },
+    { value: 4, label: 'Last Month' },
+    { value: 12, label: 'Last Quarter' },
+    { value: 52, label: 'Last Year' },
+  ];
+
+  const dateChangeHandler = (selectedDateOffset: number | string) => {
+    if (selectedDateOffset === 'Custom') {
+      setShowCustomDatePicker(true);
+    }
+    setEndDate(TODAY_DATE);
+    setStartDate(
+      subWeeks(
+        TODAY_DATE,
+        typeof selectedDateOffset === 'number' ? selectedDateOffset : 0,
+      ),
+    );
+  };
+
   return (
     <Box
       paddingY={{ base: '4', md: '8' }}
       paddingX={{ base: '4', md: '8' }}
       color="gray.700"
       width="100%"
     >
-      <PageHeading>{daoName}</PageHeading>
+      <Flex
+        direction={{ base: 'column', lg: 'row' }}
+        justifyContent="space-between"
+        alignItems="center"
+      >
+        <PageHeading>{daoName}</PageHeading>
+        <Flex
+          flexBasis={{ base: '100%', lg: '60%' }}
+          direction={{ base: 'column', lg: 'row' }}
+          alignItems="center"
+          justifyContent={{ base: 'flex-start', lg: 'flex-end' }}
+          gap={2}
+        >
+          <Flex
+            direction={{ base: 'column', lg: 'row' }}
+            alignItems="center"
+            justifyContent={{ base: 'flex-start', lg: 'flex-end' }}
+            width="100%"
+            flexBasis={{ base: '100%', lg: '60%' }}
+            flexGrow="1"
+            gap={{ base: 0, lg: 2 }}
+          >
+            <Box visibility={showCustomDatePicker ? 'inherit' : 'hidden'}>
+              <ControlledDatePicker
+                selected={startDate}
+                onChange={dates => {
+                  const datesArray = dates as Date[];
```

Comment:
Yeah -- I had some difficulty with this. The onChange is an event, but since this is a date range from the date picker the values for start and end are coming in as an array of Dates. I tried extending the date type to allow for this, but had some conflicts coming from DatePicker's types. This doesn't feel quite right to me either, but the `dates` is the synthetic event and then the `datesArray` is an array of date objects.

Reply:
Definitely feels off -- I can revisit this but I had some trouble with this due to then mismatching with the React DatePicker types

Reply:
Digging into it a bit more -- without casting the dates as the Date[] there is an issue with the type not having an iterator. I've fixed this before by spreading, but that doesn't seem to be working here.

Reply:
hm, yea, I think this is fine after looking at the DatePicker props I think they have a special signature like you were saying for the onChange prop..